lib/ical/helpers.js

```diff
@@ -287,11 +287,22 @@ ICAL.helpers = {
    */
   foldline: function foldline(aLine) {
     var result = "";
-    var line = aLine || "";
-
+    var line = aLine || "", pos = 0, line_length = 0;
+    //pos counts position in line for the UTF-16 presentation
+    //line_length counts the bytes for the UTF-8 presentation
     while (line.length) {
-      result += ICAL.newLineChar + " " + line.substr(0, ICAL.foldLength);
-      line = line.substr(ICAL.foldLength);
+      var cp = line.codePointAt(pos);
+      if (cp < 128) ++line_length;
```

Comment:
Can you run the linter here? I believe I have it set to use {} and newlines for all if statements instead of keeping them on one line.

Asde from that this seems reasonable to me, given there are tests I'm good with merging this.

Reply:
The only problem eslint had, were missing terminating semi-colons.